src/scheduler/gc_work.rs

```diff
@@ -182,33 +182,12 @@ impl<E: ProcessEdgesWork> GCWork<E::VM> for StopMutators<E> {
         trace!("stop_all_mutators start");
         mmtk.plan.base().prepare_for_stack_scanning();
         <E::VM as VMBinding>::VMCollection::stop_all_mutators(worker.tls, |mutator| {
+            // TODO: The stack scanning work won't start immediately, as the `Prepare` bucket is not opened yet (the bucket is opened in notify_mutators_paused).
+            // Should we push to Unconstrained instead?
```

Comment:
Someone may want some work to be done before scanning any stacks, but currently there is no such things.  For Ruby, "potential pinning parents" (PPP) needs to be handled before transitive closure, but can be done at the same time as stack scanning.

We also had some issue letting OpenJDK report the yielding of individual threads.  See https://github.com/mmtk/openjdk/pull/9  There used to be something called `report_java_thread_yield`, but we removed that.  So the current status is that OpenJDK can only report that all threads have come to a stop.

I think we can leave the code as is for now.  After we figure out how to report the yielding of each thread (for OpenJDK 11, 17 and 22), we can make another attempt and move the `ScanStackRoot` work packet to `Unconstrained` or another bucket.

Reply:
There is the `prepare_mutator` function for VM-specific behaviour. But this PR would remove that.

Reply:
> Someone may want some work to be done before scanning any stacks, but currently there is no such things. For Ruby, "potential pinning parents" (PPP) needs to be handled before transitive closure, but can be done at the same time as stack scanning.

The binding can do whatever work is needed before it tells MMTk that the thread is ready.

Reply:
> There is the `prepare_mutator` function for VM-specific behaviour. But this PR would remove that.

Yeah. I think it is confusing about when we should invoke it. For the old timing where we call `prepare_mutator`, the binding can do it before returning `stop_all_mutators`. JikesRVM is the only binding we know that uses the method. I changed it to https://github.com/mmtk/mmtk-jikesrvm/pull/147/files#diff-54f4f04a18788baf9bf66cff8c4256151b680087e3490fb8e8f62e2d60ad9fabR30.